src/pid-file/src/lib.rs

```diff
@@ -175,3 +183,48 @@ impl fmt::Display for Error {
 }
 
 impl std::error::Error for Error {}
+
+/// An extension of `PidFile` that also stores metadata about a process' ports
+///
+/// This is not meant to be used in production, it is to help orchestrate
+/// processes on local deployments.
+pub struct DevelopmentPidFile(PidFile);
+
+/// Contents of a `DevelopmentPidFile`
+pub struct DevelopmentPidContents {
+    /// Process PID
+    pub pid: u32,
+    /// Process port mappings
+    pub port_metadata: HashMap<String, i32>,
+}
+
+impl DevelopmentPidFile {
+    /// Attempts to open and lock the specified PID file, as well as write metadata
+    /// to the file.
+    ///
+    /// If the file is already locked by another process, it returns
+    /// `Error::AlreadyRunning`.
+    pub fn open<P: AsRef<Path>>(path: P, port_metadata: &str) -> Result<DevelopmentPidFile, Error> {
+        let pid_file = PidFile::open(&path)?;
+        let mut file = OpenOptions::new().write(true).append(true).open(path)?;
+        write!(file, "\n{port_metadata}")?;
```

Comment:
Wait, does this not break the PID file logic itself? If you were to e.g. restart computed. It seems like `pidfile_read` will read the first 16 bytes of the file, and it'll probably trip over this metadata?

Reply:
I don't think it would fail. `pidfile_read` uses `strtol` to parse the pid, which will only parse the initial part of the string that is numeric. You can take a look at this for as an example: https://godbolt.org/z/Wh6YWG46j

Though I do think that two files could be a cleaner approach.

Reply:
Ahh, of course. Gotcha!